src/platform/Darwin/UserDefaults.mm

```diff
@@ -0,0 +1,44 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+
+#include <lib/support/logging/CHIPLogging.h>
+#include <lib/support/SafeInt.h>
+#include <optional>
+
+static NSString * const kUserDefaultDomain = @"org.csa-iot.matter.darwin";
+static NSString * const kSRPTimeoutInMsecsUserDefaultKey = @"SRPTimeoutInMSecsOverride";
+
+namespace chip {
+namespace Platform {
+
+    std::optional<uint16_t> GetUserDefaultDnssdSRPTimeoutInMSecs()
```

Comment:
Note that you could also just return a `std::optional<System::Clock::Millsiconds16>` and then the name of the function does not have to encode the type....  But either way.

Reply:
will pass on this one for now :D but yeah good to keep in mind for the types